public/game.js

```diff
@@ -1,10 +1,20 @@
 const buildClearStage = ctx => () =>
   ctx.fillRect(0, 0, ctx.canvas.width, ctx.canvas.height)
 
+const centerPosToTopLeft = x => resource => x - resource.canvas.width / 2
+
+const drawObject = ctx => resource => dx => dy =>
+  ctx.drawImage(resource, centerPosToTopLeft(dx, resource), centerPosToTopLeft(dy, resource))
+
+const drawPlayer = ctx => resources => dx => dy =>
+  console.log('yolo!') ||
+  drawObject(ctx, resources.player, dx, dy)
+
 // buildImports :: CanvasRenderingContext2D -> Object
 // the functions to import into the wasm
 const buildImports = (ctx, resources) => ({
   clear_stage: buildClearStage(ctx),
+  draw_player: drawPlayer(ctx, resources),
```

Comment:
warning : the fn definition only takes 1 param (currified function). maybe refactor this fn like this :
```
const drawPlayer = (ctx, resource) => (x, y) => ctx.drawImage(/* ... */)
```